src/deparse.c

```diff
@@ -239,11 +212,20 @@ multicorn_foreign_expr_walker(Node *node,
 				if (schema != PG_CATALOG_NAMESPACE)
 					return false;
 
-				/* Make sure the specific function at hand is shippable
+                /* Make sure the specific function at hand is shippable
                  * NB: here we deviate from standard FDW code, since the allowed
                  * function list is fetched from the Python FDW instance
                  */
-				if (!list_member(fpinfo->agg_functions, makeString(opername)))
+                if (agg->aggstar)
```

Comment:
Good point bringing `COUNT(1)` up. I see this doesn't get pushed down, so I'll need to investigate a bit (and neither does the equivalent `SELECT COUNT(es.account.*) FROM es.account`) though they would be used rarely I imagine.

However note that `COUNT(some_col)` [is not equivalent to](https://www.postgresql.org/docs/12/functions-aggregate.html) `COUNT(*)`, as the former discards `NULL` values while the latter takes them into account.

Reply:
Ok, the reason `COUNT(1)` doesn't get pushed down is because `1` is treated as an expression, in particular a node of type `Const` which currently is [not being parsed](https://github.com/splitgraph/Multicorn/blob/agg-pushdown-v2-cu-1z461e4/src/deparse.c#L138). So full record fetch is performed and the counting is done on PG side.

I could add a case for parsing this node type now which is not much work in itself, but potentially some work handling edge cases since this increases the pushdown surface so to speak. Alternatively, I can just add a test for `COUNT(1)` as a reminder to have that handled at some point later on.

Reply:
Ah, yep, `COUNT(some_col)` makes sense. Adding a test for COUNT(1) works.